File: app/src/main/java/info/nukoneko/cuc/android/kidspos/entity/Staff.java
package info.nukoneko.cuc.android.kidspos.entity;

public final class Staff {
    private String barcode;
    private String name;

    @SuppressWarnings("unused")
    public Staff() {
        // for Gson
    }

    /**
     * for fake
     *
     * @param barcode barcode
     */
    public Staff(String barcode) {
        this.barcode = barcode;
        this.name = "FakeStaff";
    }

    public String getBarcode() {
        return barcode;
    }

    public String getName() {
        return name;
    }
}

File: app/src/main/kotlin/info/nukoneko/cuc/android/kidspos/entity/Staff.kt
package info.nukoneko.cuc.android.kidspos.entity

import kotlinx.serialization.Serializable

@Serializable
data class Staff(val barcode: String, val name: String) {
    companion object {
        fun create(barcode: String): Staff {
            return Staff(barcode, "Dummy")
        }
    }
}
